### ISSUE TITLE
[BUG]: Per-game "Start Fullscreen" setting doesn't work for everyone

### ISSUE THREAD

**Issue by bslenul** [PCSX2/pcsx2#7601]
### Describe the Bug

Hey!

As someone mentioned on Discord few days ago, the "Start Fullscreen" setting doesn't work on a per-game basis. So having it disabled globally but enabled per-game, the game will still start windowed. And vice-versa, having it enabled globally and disabled per-game, the game will still start fullscreen.

I checked the game .ini and it properly contains

> [UI]
StartFullscreen = true

Worth noting that some people on Discord were able to reproduce but it seems to work for others, so maybe it's conflicting with another setting somehow? Or OS specific (~~only tried on Windows 10~~ **edit:** happening on my Linux Mint VM as well)? I tried on a fresh PCSX2 install and it wasn't working for me with default settings.

### Reproduction Steps

1. Disable "Start Fullscreen" globally.
2. Right click any game > Properties > Interface and enable "Start Fullscreen".
3. Start the game.

### PCSX2 Revision

1.7.3726

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

Linux Mint 21.1 VM

**Comment by bslenul** [comment]
Finally reinstalled my Linux Mint VM (21.1 if it matters) and I can reproduce on it too.

**Comment by stenzek** [comment]
Working as expected tbh, the game settings layer isn't loaded when the VM first starts (the BIOS loads the game first), and the start full screen setting applies when the VM initializes.

So, we should either disable the setting in game settings, or pre-lookup the game settings ini when initializing the VM.... the latter seems a bit overkill just for one setting, but I guess it could potentially be used for per-game BIOS too.

**Comment by bslenul** [comment]
> the game settings layer isn't loaded when the VM first starts (the BIOS loads the game first), and the start full screen setting applies when the VM initializes.

Ah I see, makes sense then. Although now I'm even more confused about one of the devs on Discord not being able to reproduce the issue back when it was mentioned :D Probably just a misunderstanding.

> we should either disable the setting in game settings, or pre-lookup the game settings ini when initializing the VM....

I have no idea how much work that would require but yeah, might not be worth :/ Would there be any use case for per-game BIOS? 🤔 

Thanks for the reply/explanation! 👍

**Comment by stenzek** [comment]
> Would there be any use case for per-game BIOS? 🤔

Maybe not per-game, but per-region, yes - full booting only works if the disc matches the BIOS region, due to software checks.

**Comment by bslenul** [comment]
Ah OK, I thought PCSX2 was already switching BIOS automatically depending on the game region. I'm a Fast Boot user so not affected by this, but I know some people love that PS2 boot animation, so it could be nice for the purists :D

**Comment by dftf-stu** [comment]
Apologies if this isn't related... but I looked-through existing tickets and found 11954 , which sounded like the issue I have, and that one was closed as a duplicate of this one.

I'm essentially having two issues in the latest PCSX2 (v2.3.423, as of the time of this comment):

- I cannot set any custom resolutions for my laptop's Intel iGPU

> - The only option is "Borderless Fullscreen"
> - I have tried two suggestions I've found online that are supposed to get them back, but neither worked:
> - Settings > Interface > Enable the "Render To Separate Window" option
> - Settings > Graphics > Advanced tab > change "Allow Exclusive Fullscreen" to a different setting

- I _can_ set custom resolutions for my laptop's discreet AMD GPU, but the output doesn't actually change

> - If I pick my discreet GPU as the "Adapter" _then I can_ pick custom-resolutions in the "Fullscreen Mode:" dropdown
> - However... none of them seem to actually change anything.  The lowest, 640 x 480, should look noticeably pixelated compared to the native-resolution, but doesn't look any-different. 🤷🏻‍♂️

On my older DirectX 10-only laptop, PCSX2 v1.7.4832 (the final release that device can use) allows me to choose any resolution its iGPU supports, _and_ the output actually does change when I select a different option.

Is there something-else I need to do in the v2.x series for this to work (change some-other setting?), or are other-resolutions just unsupported and only the resolution the display PCSX2 is currently running on can be used?

**Comment by RedPanda4552** [comment]
You are referencing the display mode overrides that Direct3D uses. These are not present on any other renderer, and Direct3D should not be used except for very old systems where both Vulkan and OpenGL are not supported. There is no utility to these modes; if you want to upscale, that is handled with internal resolution, not altering your display resolution.

Edit: bad close

**Comment by dftf-stu** [comment]
@RedPanda4552: according to GPU-Z:
- iGPU: OpenGL 4.6, DirectX 12(11_1), WDDM 2.0, no Vulcan support
- dGPU: OpenGL 4.6, DirectX 12(11_1), WDDM 2.7, has Vulcan support

I've never had any graphical-issues running games on the iGPU, and it usually uses less-power and runs cooler.  I was just looking
to output at a _lower_ resolution, to reduce stress.  If I'm running a PS2 game that's around 480p native, then I could set 640 x 480
as the full-screen resolution to lower the iGPU load.  But it seems that's not possible now?

I wasn't looking to upscale (I'm aware that's done at: Settings > Graphics > Rendering > Internal Resolution).

So... in-short, is it not supported now to have PCSX2 switch to a lower-resolution in full-screen than what Windows is set to?

**Comment by RedPanda4552** [comment]
No, it is not supported to switch screen resolution. Frankly, if screen resolution is significant enough to impact performance, that's a greater sign that your device is severely underpowered and not suitable to run PCSX2. Your screen resolution has no bearing on how hard PCSX2 works the GPU to render output, so if it is making an impact in your case that speaks to a much more significant issue with your hardware being insufficient to keep up with the Windows compositor even.
